src/heuristic.c

```diff
@@ -0,0 +1,221 @@
+ * GNU General Public License for more details.
+ */
+
+/**
+ * @file heuristic.c
+ * @author Scott Chu
+ * @date 2020-05-28
+ */
+
+/*File Includes*/
+#include "../inc/heuristic.h"
+#include "cloud_detection.c"
+
+/*Standard Lib Includes*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <math.h>
+
+struct input_feature img_buffer[10]; //Temporary solution for storing image parameters, this will eventually be stored in a buffer with the images.
+
+/**
+ * @brief
+ * 		Generates an dummy image with a "width" amount of pixels in the x-direction.
+ * @param width
+ * 		Amount of pixels in the x-direction.
+ * @return
+ * 		Returns 1 if not sucessful.
+ */
+int generate_img(int width)
+{       
+        //Create image raster file 
+        FILE *fptr;
+        fptr = fopen("../test/testImage.bin", "wb");
+        if(fptr == NULL){
+                return 1;
+        }
+
+        srand(0);
+
+        //Dummy input parameter information
+        input_feature_t input_params;
+        input_params.signed_samples = 1;
+        input_params.dyn_range = 8;
+        input_params.x_size = width;
+        input_params.y_size = Y_PIXELS;
+        input_params.z_size = Z_PIXELS;
+        input_params.in_interleaving = BSQ;
+        input_params.in_interleaving_depth = 1; 
+        input_params.byte_ordering = LITTLE;
+        input_params.regular_input = 1;
+        img_buffer[0] = input_params;
+
+        //Generating random sample band values in specified raster formats
+        if (input_params.in_interleaving == BSQ) {
+                for (int z = 0; z < input_params.z_size; z++) {
+                        for (int y = 0; y < input_params.y_size; y++){
+                                for (int x = 0; x < input_params.x_size; x++) {
+                                        int sample = rand() % (256 + 1 - 0) + 0;
+                                        fprintf(fptr, "%d ", sample);
+                                }
+                                fprintf(fptr, "\n");
+                        }
+                }       
+        } else if (input_params.in_interleaving == BIP || input_params.in_interleaving == BIL) {
+                for (int y = 0; y < input_params.y_size; y++) {
+                        for (int x = 0; x < input_params.x_size * input_params.z_size; x++) {
+                                int sample = rand() % (256 + 1 - 0) + 0;
+                                fprintf(fptr, "%d ", sample);
+                        }
+                        fprintf(fptr, "\n");
+                }                  
+        }
+
+        fclose(fptr);
+}
+
+/**
+ * @brief
+ * 		Calculates the heuristic value of an image and stores it in image header.
+ * @param imgnum
+ * 		Designator for the image.
+ * @return
+ * 		Returns 1 if not sucessful.
+ */
+int heuristic_img(int imgnum)
+{
+        int raw_heuristic = 0;
+        FILE *fptr;
+        fptr = fopen("../test/testImage.bin", "rb");
```

Comment:
We'll also want to make sure that all this stuff works with RelianceEdge. I know almost nothing about RelianceEdge, but these file functions might be slightly different. If the are though it might be worthwhile to just make a wrapper library so that the functions look the same as these unix file functions.